sentinelhub/areas.py

```diff
@@ -475,3 +484,152 @@ def _make_split(self):
 
                         self.bbox_list.append(bbox)
                         self.info_list.append(info)
+
+
+class BaseUtmSplitter(AreaSplitter):
+    """ Base splitter that returns bboxes of fixed size aligned to UTM zones or UTM grid tiles as defined by the MGRS
+    """
+    def __init__(self, shape_list, crs, bbox_size):
+        """
+        :param shape_list: A list of geometrical shapes describing the area of interest
+        :type shape_list: list(shapely.geometry.multipolygon.MultiPolygon or shapely.geometry.polygon.Polygon)
+        :param crs: Coordinate reference system of the shapes in `shape_list`
+        :type crs: CRS
+        :param bbox_size: Physical size in metres of generated bounding boxes. Could be a float or tuple of floats
+        :type bbox_size: int or (int, int) or float or (float, float)
+        """
+        super().__init__(shape_list, crs)
+
+        self.bbox_size = self._parse_split_parameters(bbox_size, allow_float=True)
+
+        self.shape_geometry = Geometry(self.area_shape, self.crs).transform(CRS.WGS84)
+
+        self.utm_grid = self._get_utm_polys()
+
+        self._make_split()
+
+    @abstractmethod
+    def _get_utm_polys(self):
+        raise NotImplementedError
+
+    @staticmethod
+    def _get_utm_from_props(utm_dict):
+        """ Return the UTM CRS corresponding to the UTM described by the properties dictionary
+
+        :param utm_dict: Dictionary reporting name of the UTM zone and MGRS grid
+        :type utm_dict: dict
+        :return: UTM coordinate reference system
+        :rtype: sentinelhub.CRS
+        """
+        return CRS('32{}{}'.format(6 if utm_dict['direction'] == 'N' else 7, str(utm_dict['zone']).zfill(2)))
+
+    def _align_bbox_to_size(self, bbox):
+        """ Align input bbox coordinates to be multiples of the bbox size
+
+        :param bbox: Bounding box in UTM coordinates
+        :type bbox: sentinelhub.BBox
+        :return: BBox objects with coordinates multiples of the bbox size
+        :rtype: sentinelhub.BBox
+        """
+        size_x, size_y = self.bbox_size
+        lower_left_x, lower_left_y = bbox.get_lower_left()
+        return BBox([(lower_left_x - lower_left_x % size_x, lower_left_y - lower_left_y % size_y),
+                     bbox.get_upper_right()], crs=bbox.crs)
+
+    def _make_split(self):
+        """ Split each UTM grid into equally sized bboxes in correct UTM zone
+        """
+        size_x, size_y = self.bbox_size
+        self.bbox_list = []
+        self.info_list = []
+
+        index = 0
+
+        for utm_cell in self.utm_grid:
+            utm_cell_geom, utm_cell_prop = utm_cell
+            if utm_cell_prop['zone'] == 0:
+                continue
+            utm_crs = self._get_utm_from_props(utm_cell_prop)
+
+            intersection = utm_cell_geom.intersection(self.shape_geometry.geometry)
+
```

Comment:
Actually, this intersection could be a `GeometryCollection` (in some very rare cases) therefore I would add the following:
```Python
if not intersection.is_empty and isinstance(intersection, GeometryCollection):
    intersection = MultiPolygon(geo_object for geo_object in intersection if isinstance(geo_object, (Polygon, MultiPolygon))

if not intersection.is_empty:
    ... 
```
That way you filter out various points and line strings from the intersection.

Reply:
changed